hwy/ops/arm_neon-inl.h

```diff
@@ -6484,6 +6484,406 @@ HWY_API VFromD<D> SlideDownLanes(D d, VFromD<D> v, size_t amt) {
   return detail::SlideDownLanes(v, amt);
 }
 
+// ------------------------------- WidenHighMulAdd
+
+#ifdef HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#undef HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#else
+#define HWY_NATIVE_WIDEN_HIGH_MUL_ADD
+#endif
+
+namespace detail {
+
+template<class D, HWY_IF_U32(TFromD<D>), HWY_IF_LANES_GT_D(D, 2),
```

Comment:
Nice you've got the SFINAE to handle (disable) the case of N=1 inputs :)
We can simplify a bit to `HWY_IF_U32_D(D)`.
One remaining issue: there is a consistent convention that D describes the return type. Would you mind changing all functions to 
```
template<class D, HWY_IF_Uxx(D), HWY_IF_LANES_GT_D(D, 2),
class DN = RepartitionToNarrow<D>>
HWY_API VFromD<D> WidenHighMulAdd(D /* tag */, VFromD<DN> mul,
                                   VFromD<DN> x, VFromD<D> add) {
```
? So D is the wide, and DN is the narrow input.

Reply:
Got it. I'll make changes